crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -1127,40 +1127,49 @@ impl SlidingSync {
             }
         };
 
-        let request = assign!(v4::Request::new(), {
-            lists: requests,
-            pos,
-            delta_token,
-            timeout: Some(timeout),
-            room_subscriptions,
-            unsubscribe_rooms,
-            extensions: extensions.clone().unwrap_or_default(),
-        });
-        debug!("requesting");
-
-        // 30s for the long poll + 30s for network delays
+        debug!("Sending the sliding sync request");
+
+        // Configure long-polling. We need 30 seconds for the long-poll itself, in
+        // addition to 30 more extra seconds for the network delays.
         let request_config = RequestConfig::default().timeout(timeout + Duration::from_secs(30));
+
+        // Prepare the request.
         let request = self.client.send_with_homeserver(
-            request,
+            assign!(v4::Request::new(), {
+                lists: lists_of_requests,
+                pos,
+                delta_token,
+                timeout: Some(timeout),
+                room_subscriptions,
+                unsubscribe_rooms,
+                extensions: extensions.clone().unwrap_or_default(),
+            }),
             Some(request_config),
             self.homeserver.as_ref().map(ToString::to_string),
         );
 
+        // Send the request and get a response with end-to-end encryption support.
```

Comment:
Perhaps we could be a bit more verbose and mention the issue here that we intruduced:

```rust
// Sending the `/sync` request out when end-to-end encryption is enabled means that
// we need to also send out any outgoing e2ee related request out coming from the
// `OlmMachine::outgoing_requests()` method.
//
// FIXME: Processing outgiong requests at the same time while a `/sync` is in
// flight is currently not supported.
// More info: [#1386](https://github.com/matrix-org/matrix-rust-sdk/issues/1386).

Reply:
:+1: